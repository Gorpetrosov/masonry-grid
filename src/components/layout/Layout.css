.layout {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    width: 100%;
    margin: 0 auto;
}

.layout-header {
    background-color: var(--primary-color);
    padding: var(--header-padding);
    color: white;
    text-align: center;
    font-size: var(--header-font-size);
    font-weight: 600;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    position: relative;
    z-index: 100;
}

.layout-footer {
    background-color: var(--primary-color);
    padding: var(--footer-padding) 0;
    color: white;
    text-align: center;
    font-size: var(--footer-font-size);
    margin-top: auto;
    width: 100%;
    box-shadow: 0 -4px 6px rgba(0, 0, 0, 0.1);
}

@media screen and (max-width: 768px) {
    .layout-header {
        padding: 15px;
        font-size: 1.1rem;
    }

}

@media screen and (max-width: 480px) {
    .layout {
        min-height: 100svh;
    }

    .layout-header {
        padding: 10px;
        font-size: 1rem;
    }

    .layout-footer {
        font-size: 0.7rem;
        padding: 8px 0;
    }
}