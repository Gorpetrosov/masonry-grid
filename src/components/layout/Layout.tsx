import {ReactNode} from 'react';
import config from "../../config";
import "./Layout.css"

interface LayoutProps {
    children: ReactNode;
    headerText?: string;
}

const currentYear = new Date().getFullYear();

export const Layout = ({ children, headerText = "Welcome Masonry Grid"}: LayoutProps) => {
    return (
        <div className="layout">
            <header className="layout-header">
                <h1>{headerText}</h1>
            </header>
            <div className="layout-main">{children}</div>
            <footer className="layout-footer">
                <p>Â© {currentYear} {config.APP_NAME}</p>
            </footer>
        </div>
    );
};

export default Layout;
