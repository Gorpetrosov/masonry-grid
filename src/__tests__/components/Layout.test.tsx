import { render, screen } from '@testing-library/react';
import { describe, test, expect , vi} from 'vitest';
import { MemoryRouter } from 'react-router-dom';
import Layout from "../../components/layout/Layout.tsx";


vi.mock('../../components/Header.tsx', () => ({
    default: ({ headerText }: { headerText: string }) => (
        <header data-testid="header">{headerText}</header>
    )
}));

vi.mock('../../components/Footer.tsx', () => ({
    default: () => <footer data-testid="footer" />
}));

describe('Layout Component', () => {

    afterEach(() => {
        vi.resetAllMocks()
    });

    const TestChild = () => <div data-testid="test-child">Content</div>;

    test('renders header, main content, and footer', () => {
        render(
            <MemoryRouter>
                <Layout>
                    <TestChild />
                </Layout>
            </MemoryRouter>
        );

        expect(screen.getByTestId('header')).toBeInTheDocument();
        expect(screen.getByTestId('test-child')).toBeInTheDocument();
        expect(screen.getByTestId('footer')).toBeInTheDocument();
    });

    test('uses default header text when not provided', () => {
        render(
            <MemoryRouter>
                <Layout children={undefined} />
            </MemoryRouter>
        );

        const header = screen.getByTestId('header');
        expect(header).toHaveTextContent('Welcome Masonry Grid');
    });

    test('uses custom header text when provided', () => {
        const customText = 'Custom Header Text';
        render(
            <MemoryRouter>
                <Layout headerText={customText} children={undefined} />
            </MemoryRouter>
        );

        const header = screen.getByTestId('header');
        expect(header).toHaveTextContent(customText);
    });

    test('applies correct layout structure', () => {
        const { container } = render(
            <MemoryRouter>
                <Layout children={undefined} />
            </MemoryRouter>
        );

        const layoutDiv = container.querySelector('.layout');
        expect(layoutDiv).toBeInTheDocument();
        expect(layoutDiv?.querySelector('.layout-main')).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { container } = render(
            <MemoryRouter>
                <Layout headerText="Test Header">
                    <TestChild />
                </Layout>
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});