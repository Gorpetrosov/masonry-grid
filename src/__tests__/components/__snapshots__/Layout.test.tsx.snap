// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout Component > matches snapshot 1`] = `
<div>
  <div
    class="layout"
  >
    <header
      data-testid="header"
    >
      Test Header
    </header>
    <div
      class="layout-main"
    >
      <div
        data-testid="test-child"
      >
        Content
      </div>
    </div>
    <footer
      data-testid="footer"
    />
  </div>
</div>
`;
